content/post/fewer-parameters-means-fewer-bugs.md

```diff
@@ -0,0 +1,244 @@
+---
+title: "Fewer parameters means fewer bugs"
+date: 2021-07-13T22:57:39+09:00
+hero: /images/hero-5.jpg
+excerpt: "Tips for coding efficiently and reducing bugs"
+timeToRead: 10
+authors:
+  - Bater
+draft: true
+---
+When I review code, I sometimes notice that there is an opportunity to squash method parameters - yet many developers are unaware of this when they write code. From my coding experience, we can write code more efficiently and reduce the amount of bugs by making an effort to reduce the number of parameters in our methods. Unless your manager still evaluates output by lines of code, you should definitely try these tips in your daily coding life.
+
+So today, I will share how to do it with "OOP" (object-oriented programming) design, and explain where this magic power comes from.
+
+Let's start with a small example. The following demo code was written in Ruby, but don't worry; Ruby code is so beginner-friendly that once you have some "OOP" background, it's easy to grasp.
+
+The example story is, we have some books and want to pass the author name as a parameter, and get the book title, author name, and year published as a summary.
+```ruby
+class Book
+  attr_reader :title, :year_published
+ 
+  def initialize(title, year_published)
+    @title = title
+    @year_published = year_published
+  end
+ 
+  def summary(first_name, last_name)
+    "#{title} - #{last_name}, #{first_name} (#{year_published})"
+  end
+end
+ 
+book = Book.new("Ruby is the best", "2020")
+book.summary("Bater", "Chen")
+=> "Ruby is the best - Chen, Bater (2020)"
+```
+We have a book class. It has two attributes: `title`, and `year_published`. The `initialize` method is called when we `new` a book object. Book class only has one instance method called `summary`. It needs two parameters: first_name and last_name. In Ruby, when we define the `attr_reader` as the attribute of the object, it will create the read method automatically. So we can call the instance variable directly by `title` to get `@title` return, same as `year_published`.
```

Comment:
```suggestion
We have a book class. It has two attributes: `title`, and `year_published`. The `initialize` method is called when we `new` a book object. Book class only has one instance method called `summary`. It needs two parameters: `first_name` and `last_name`. In Ruby, when we define the `attr_reader` as the attribute of the object, it will create the read method automatically. So we can call the instance variable directly by `title` to get `@title` return, same as `year_published`.
```